File: src/frontend/services/analytics.service.ts
import { apiClient } from "@/lib/api-client";

export interface AnalyticsSummary {
  totalSkills: number;
  completedSteps: number;
  totalSteps: number;
  trend: { date: string; value: number }[];
}

export const analyticsService = {
  summary: () => apiClient.get<AnalyticsSummary>("/analytics/summary").then((r) => r.data),
};

File: src/backend/app/services/analytics_service.py
class AnalyticsService:
    async def summary(self) -> dict:
        # TODO: compute real metrics
        return {
            "totalSkills": 0,
            "completedSteps": 0,
            "totalSteps": 0,
            "trend": [],
        }


analytics_service = AnalyticsService()
